eos-tech-support/eos-reset-password

```diff
@@ -31,14 +31,49 @@ else
 fi
 
 MOUNT=`mktemp -d`
+LOOPBACK=
 
 cleanup() {
     set +e
-    umount $MOUNT
-    rmdir $MOUNT
+
+    umount "$MOUNT"
+
+    if [ -n "$LOOPBACK" ]; then
+        losetup -d "$LOOPBACK"
+        udevadm settle
+
+        umount "$MOUNT"
+    fi
+
+    rmdir "$MOUNT"
 }
 trap cleanup EXIT
 
 mount $DEVICE $MOUNT
+
+# Check if it's a Windows partition with a dual-boot Endless OS image
+LABEL=$(lsblk -o LABEL --raw --nodeps --noheadings "$DEVICE")
+ENDLESS_IMG="$MOUNT/endless/endless.img"
+if [ "$LABEL" != "ostree" ] && [ -e "$ENDLESS_IMG" ]; then
```

Comment:
I think you should check whether `$LABEL` is an empty string to protect against bogus runs of `lsblk`